components/Dialogs/src/MDCAlertController.m

```diff
@@ -186,6 +189,48 @@ - (void)addAction:(MDCAlertAction *)action {
   [self.view setNeedsLayout];
 }
 
+- (BOOL)mdc_adjustsFontForContentSizeCategory {
+  return _mdc_adjustsFontForContentSizeCategory;
+}
+
+- (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)adjusts {
+  _mdc_adjustsFontForContentSizeCategory = adjusts;
+
+  // Update any action buttons to match the alert controller.
+  for (MDCButton *actionButton in self.actionButtons) {
+    actionButton.mdc_adjustsFontForContentSizeCategory = adjusts;
+  }
+
+  if (_mdc_adjustsFontForContentSizeCategory) {
+    [self updateFontsForDynamicType];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(contentSizeCategoryDidChange:)
+                                                 name:UIContentSizeCategoryDidChangeNotification
+                                               object:nil];
+  } else {
+    [[NSNotificationCenter defaultCenter] removeObserver:self
+                                                    name:UIContentSizeCategoryDidChangeNotification
+                                                  object:nil];
+  }
+}
+
+// Handles UIContentSizeCategoryDidChangeNotifications
+- (void)contentSizeCategoryDidChange:(NSNotification *)notification {
+  [self updateFontsForDynamicType];
+}
+
+// Update the fonts used based on mdc_preferredFontForMaterialTextStyle and recalculate the
+// preferred content size.
+- (void)updateFontsForDynamicType {
+  self.titleLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:MDCFontTextStyleTitle];
```

Comment:
I do believe we should land this as is.
This bool property is based off of the property added to label, text view and text field; adjustsFontForContentSizeCategory / this is the way Apple has chosen to expose it.
One of the reasons we are recommending system fonts is because of the difficulty in supporting custom fonts and dynamic type.  
If we chose to expose custom fonts here, clients will end up breaking dynamic type unless they are incredibly careful, have properly hinted fonts, and do extensive testing.
By not exposing that capability we stop all of those bugs before they start.
tl;dr: I don't believe we should support custom fonts in MDCAlertController at this time.